crates/wasmtime/src/component/types.rs

```diff
@@ -658,7 +658,8 @@ fn variant_size_and_alignment(types: impl ExactSizeIterator<Item = Type>) -> Siz
     }
 
     SizeAndAlignment {
-        size: func::align_to(usize::from(discriminant_size), alignment) + size,
+        size: func::align_to(usize::from(discriminant_size), alignment)
+            + func::align_to(size, alignment),
```

Comment:
Similar comment as to above about aligning the result of the summation instead of each individual component to match the spec (although again I don't think it affects the results)